### PR TITLE
feat: changesets for the hybrid removal

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

It's all changesets for https://github.com/withastro/astro/pull/11824

## Testing

N/A

## Docs

N/A

### CODE REVIEW

**Review by matthewp — APPROVED** [review]

File: .changeset/afraid-apricots-buy.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      name: 'my-adapter',
+      adapterFeatures: {
+        buildOutput: 'server',
+      },
+    });
+  },
+}
+```
+
+If your adapter specify `buildOutput: 'static'`, and the user's project contains server-rendered pages, Astro will warn in development and error at build time. Note that an hybrid output, containing both static and server-rendered pages, is considered to be a `server` output, as a server is required to serve the server-rendered pages.
```
Comment (sarah11918) [inline .changeset/afraid-apricots-buy.md:20]:
```suggestion
If your adapter specifies `buildOutput: 'static'`, and the user's project contains server-rendered pages, Astro will warn in development and error at build time. Note that a hybrid output, containing both static and server-rendered pages, is considered to be a `server` output, as a server is required to serve the server-rendered pages.
```

File: .changeset/giant-rocks-thank.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+'astro': major
+---
+
+Merges the `output: 'hybrid'` and `output: 'static'` option. It is no longer necessary to specify `output: 'hybrid'` in your Astro config to use server rendered pages. Astro will now automatically determine if the build output should be static or hybrid based on the presence of server-rendered pages in your project.
+
+If your project used hybrid rendering, you can now remove the `output: 'hybrid'` option from your Astro config. If you were using the `output: 'static'` option, you can continue to use it as before.
```
Comment (sarah11918) [inline .changeset/giant-rocks-thank.md:7]:
```suggestion
Merges the `output: 'hybrid'` and `output: 'static'` configurations into one single configuration (now called `'static'`) that works the same way as the previous `hybrid` option.

It is no longer necessary to specify `output: 'hybrid'` in your Astro config to use server-rendered pages. The new `output: 'static'` has this capability included. Astro will now automatically provide the ability to opt out of prerendering in your static site with no change to your `output` configuration required. Any page route or endpoint can include `export const prerender = false` to be server-rendered, while the rest of your site is statically-generated.

If your project used hybrid rendering, you must now remove the `output: 'hybrid'` option from your Astro config as it no longer exists. However, no other changes to your project are required, and you should have no breaking changes. The previous `'hybrid'` behavior is now the default, under a new name `'static'`.

If you were using the `output: 'static'` (default) option, you can continue to use it as before. By default, all of your pages will continue to be prerendered and you will have a completely static site. You should have no breaking changes to your project.
```

File: .changeset/giant-rocks-thank.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+```diff
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+-  output: 'hybrid',
+});
+```
+
+An adapter is still required to deploy an Astro project with server-rendered pages, failure to include an adapter will result in a warning in development and an error at build time.
```
Comment (sarah11918) [inline .changeset/giant-rocks-thank.md:17]:
```suggestion
An adapter is still required to deploy an Astro project with any server-rendered pages. Failure to include an adapter will result in a warning in development and an error at build time.
```

**Review by sarah11918 — COMMENTED** [review]
These look great @Princesseuh! I just made some tiny suggestions for you, as I think being more explicit in the major will prevent some questions people might naturally have.

**Pull request merged**